### DISCUSSION TITLE
Error (Xcode): Multiple commands produce "...GoogleUtilities_Privacy.bundle'

### DISCUSSION THREAD

**Discussion by VeeTsien** [codemagic-ci-cd/codemagic-docs#2416] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Not sure about the steps to reproduce tbh. The build works on my local machine with the same environment. If anything, this only occurred after working on 3rd party integration with customer io and a bunch of config changes in xcode like adding a new build target for notification extension service. 

### Expected results

Build successfully

### Actual results

This is the same problem as this [thread](https://github.com/orgs/codemagic-ci-cd/discussions/2181) but since it's closed I'm opening a new one. 

This [thread](https://github.com/AngeloAvv/flutter_flavorizr/issues/46#issuecomment-968896934) seems to be the solution for a lot of people to remove the GoogleService-the target build phase Copy Bundle Resources, tho my error message isn't google service info.plist related I thought why not and give it a try, still doesn't fix it. 

Here's the error message: 

```
Error (Xcode): Multiple commands produce '/Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/GoogleUtilities_Privacy.bundle'

Error (Xcode): Multiple commands produce '/Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/GoogleUtilities.framework'

```

Any help is much appreciated!

### Build id (optional)

65f4dcfc6b900373a6b9697e

---
New info:
I'm seeing the duplicates of GoogleUtilities in pods, but my local build did not complain about it. 
<img width="1920" alt="Screenshot 2024-03-15 at 23 20 06 (2)" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/53539534/b7c48166-8a92-4ab8-96f4-4ab57694a43c">

And here is my podfile in case it helps:
[Podfile](https://gist.github.com/VeeTsien/34b0457581d48ff71e056f7f09c16fa4#file-podfile)

**Comment by VeeTsien** [comment, marked as answer]
Solved by adding `pod 'GoogleUtilities'` in the Podfile for the two different targets. 

e.g.
```
target 'NotificationServiceExtension' do
  pod 'GoogleUtilities'
end
```

Still wondering tho why it builds successfully locally but not in CodeMagic.

**Reply by unbam** [reply]
I had that set up and it did not work in Codemagic

**Reply by ChhengkheangO** [reply]
Thanks, that solved my issue. I didn't put that in target runner and only in extenstion target.

**Comment by lukemmtt** [comment]
@VeeTsien You're a hero—I'm super curious how you figured this one out.

**Comment by AvinashShine** [comment]
target 'NotificationServiceExtension' do
  pod 'GoogleUtilities'
end

did not work

**Reply by andyshephard** [reply]
Make sure that you add the pod 'GoogleUtilities' line to BOTH target sections in the Podfile.
Clean the Pods directory and delete the Podfile.lock file to ensure that you'll start fresh and then run `pod install`

Like the image below, you can then check the list of dependencies in your Pods target in Xcode. You should only see single instances of GoogleUtilities at that point

1. Duplicated GoogleUtilities dependencies
![image](https://github.com/user-attachments/assets/94638274-f4ab-4a71-8331-771ecee28082)

2. Fixed, single GoogleUtilities dependency
<img width="997" alt="Screenshot 2025-05-12 at 16 45 58" src="https://github.com/user-attachments/assets/3e61a622-e6f7-4b88-81c0-15927eb46343" />

**Comment by re2005** [comment]
I see some of you still have the same issues.

Including this line on Podfile, before you declare your target:

```
pod 'GoogleUtilities', :modular_headers => true
```

After run inside `ios` folder `pod install` 

That will make xCode to use a single version of Google Utilities.

In my case using it for Authentication, Firebase, Push Notifications etc...

I hope this helps, if you need more details I can explain everything I've done to have a successful build.

**Reply by kunalmaurya6** [reply]
Thanks! It worked for me.
I was able to build the archive successfully.
